content/en/security_platform/application_security/troubleshooting.md

```diff
@@ -0,0 +1,528 @@
+| Linux	     | /var/log/datadog/dotnet/	                                  |
+| Windows 	 | C:\ProgramData\Datadog .NET Tracer\logs                    |
+
+[1]: https://docs.docker.com/storage/volumes/
+{{< /programming-lang >}}
+{{< programming-lang lang="PHP" >}}
+
+For PHP, to start troubleshooting issues with the Datadog Application Security extension, enable debug logs in the Application Security extension’s `.ini` file.
+
+The extension's `ini` file is usually found in `/etc/php/<version>/xxx/conf.d/98-ddappsec.ini`, but the location may differ depending on your installation. Look at the beginning of the `phpinfo()` output to identify the directory that is scanned for `.ini` files, if any. In the `.ini` file, set the following configuration options with the following: 
+
+```php
+ddappsec.log_level=‘debug’
+ddappsec.helper_extra_args=‘--log_level=debug’
+ddappsec.helper_log_file=‘/tmp/helper.log’
+```
+
+The extension outputs logs to the default `php_error` log file. If there are no logs in the file, add the following to the `.ini` file:
+
+```php
+ddappsec.log_file=’tmp/extension.log’
+```
+
+### Installation fails to find PHP
+If the installation script is unable to find the correct PHP version, you can set the `--php-bin` to the PHP binary location, for example:
+
+```
+$ php dd-library-php-setup.php --php-bin /usr/bin/php7.4 --tracer-version=latest --appsec-version=latest
+```
+### Connection to helper failed
+If the Application Security extension is unable to communicate with the helper process, the following warning occurs:
+
+```
+PHP Warning:  Unknown: [ddappsec] Connection to helper failed and we are not going to attempt to launch it: dd_error
+```
+
+The warning could be followed by one of these error messages:
+
+```
+PHP Warning:  Unknown: [ddappsec] Could not open lock file /tmp/ddappsec.lock: Permission denied in Unknown on line 0
+```
+```
+PHP Warning:  Unknown: [ddappsec] Call to bind() failed: Permission denied
+```
+```
+PHP Warning:  Unknown: [ddappsec] Failed to unlink /tmp/ddappsec.sock: Operation not permitted
+```
+
+This indicates that the lock file or socket file used by the extension has invalid permissions, or the user executing the PHP process does not have write access to the `tmp` directory. 
+
+If the lock file or socket file has invalid permissions, you can either delete them and restart Apache/FPM or adjust the `user:group` to match the one used by Apache/FPM, for example, `www-data`.
+
+If the user doesn’t have write access to the tmp directory, you can change the location of the lock file and socket file by modifying the following settings in the extension’s `.ini` file:
+
+```
+ddappsec.helper_socket_path = /<directory with compatible permissions>/ddappsec.sock
+ddappsec.helper_lock_path = /<directory with compatible permissions>/ddappsec.lock
+```
+
+{{< /programming-lang >}}
+{{< programming-lang lang="go" >}}
+
```

Comment:
```suggestion

#### Knowing if AppSec is enabled in the running application

The startup logs show the tracer configuration and whether AppSec in enabled or not. AppSec is enabled when the configuration field `appsec` is `true` and disabled otherwise.

For example, the following startup log shows the AppSec is disabled:
```
2022/02/17 14:49:00 Datadog Tracer v1.36.0 INFO: DATADOG TRACER CONFIGURATION {"date":"2022-02-17T14:49:00+01:00","os_name":"Linux (Unknown Distribution)","os_version":"5.13.0","version":"v1.36.0","lang":"Go","lang_version":"go1.17.1","env":"prod","service":"grpcserver","agent_url":"http://localhost:8126/v0.4/traces","debug":false,"analytics_enabled":false,"sample_rate":"NaN","sampling_rules":null,"sampling_rules_error":"","service_mappings":null,"tags":{"runtime-id":"69d99219-b68f-4718-9419-fa173a79351e"},"runtime_metrics_enabled":false,"health_metrics_enabled":false,"profiler_code_hotspots_enabled":false,"profiler_endpoints_enabled":false,"dd_version":"","architecture":"amd64","global_service":"","lambda_mode":"false","appsec":false,"agent_features":{"DropP0s":false,"Stats":false,"StatsdPort":0}}
```

#### Enabling debug logs

Debug logs can be enabled with the environment variable `DD_TRACE_DEBUG=1`. AppSec library will log to the standard error output what is happening.

Note that AppSec only logs when it is enabled (using the environment variable `DD_APPSEC_ENABLED=1`). 
```